File: java-example/src/test/java/org/example/CalculatorIntegrationTest.java
package org.example;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class CalculatorIntegrationTest {
    private final Calculator calculator = new Calculator();

    @Test
    void testComplexCalculation() {
        // (10 + 5) * 2 = 30
        int result1 = calculator.add(10, 5);
        assertEquals(30, calculator.multiply(result1, 2));

        // (20 - 5) / 3 = 5
        int result2 = calculator.subtract(20, 5);
        assertEquals(5.0, calculator.divide(result2, 3));
    }
}

File: js-example/calculator.integration.test.js
const { describe, test, expect } = require('@jest/globals');
const Calculator = require("./calculator");

describe('Calculator Integration', () => {
    const calculator = new Calculator();

    test('complex calculations', () => {
        // (10 + 5) * 2 = 30
        const result1 = calculator.add(10, 5);
        expect(calculator.multiply(result1, 2)).toBe(30);

        // (20 - 5) / 3 = 5
        const result2 = calculator.subtract(20, 5);
        expect(calculator.divide(result2, 3)).toBe(5.0);
    });
});